qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/LargeMessageSizeTest.java

```diff
@@ -134,4 +151,64 @@ public void shouldCompleteJobWithLargeVariables() {
     // then
     ZeebeAssertHelper.assertProcessInstanceCompleted(processInstanceEvent.getProcessInstanceKey());
   }
+
+  @Test
+  public void shouldActivateJobsByRespectingMaxMessageSize() {
+    // given
+    final var modelInstance =
+        Bpmn.createExecutableProcess("foo")
+            .startEvent()
+            .serviceTask()
+            .zeebeJobType("foo")
+            .endEvent()
+            .done();
+
+    CLIENT_RULE
+        .getClient()
+        .newDeployResourceCommand()
+        .addProcessModel(modelInstance, "foo.bpmn")
+        .send()
+        .join();
+
+    final var byteArray = new byte[1024 * 1024]; // 1 MB
+    new Random().nextBytes(byteArray);
+    final var message = new String(byteArray, StandardCharsets.UTF_8);
+
+    final int numberOfJobsToActivate = 5;
+    for (int i = 0; i < numberOfJobsToActivate; i++) {
+      final ProcessInstanceEvent event =
+          CLIENT_RULE
+              .getClient()
+              .newCreateInstanceCommand()
+              .bpmnProcessId("foo")
+              .latestVersion()
+              .variables(Map.of("message_content", message))
+              .send()
+              .join();
+
+      RecordingExporter.processInstanceRecords()
+          .withProcessInstanceKey(event.getProcessInstanceKey())
+          .withElementType(BpmnElementType.PROCESS)
+          .await();
```

Comment:
🔧 I think we don't have to wait for the command to be appended to the log in the loop because sending the 5 create instance commands and processing them asynchronously should be fine. I don't think it affects the rest of this test case.

Reply:
@korthout is it because the instances themselves are lightweight so will be created fast enough? If so, could you give me some examples where we should use `await()` method? I am asking to clarify for the next time I use it. 👍

Reply:
Definitely more clear, thanks! 🚀